doc/elements-tx-format.md

```diff
@@ -0,0 +1,570 @@
+# Elements Transaction Format
+
+This document describes the format used to serialize Elements transactions. Once a transaction has been converted into this raw, serialized form, it can be broadcast across the network.
+
+This document requires familiarity with the architecture of Bitcoin and Elements, as well the various parts that are expected in an Elements transaction (inputs, outputs, assets, pegin data, etc.). For more information on those, please refer to the [Elements code tutorial](https://elementsproject.org/elements-code-tutorial/overview).
+
+### Data Types
+
+*Note:* The encoding column in the following tables corresponds to the way that the fields are serialized. In Bitcoin, numbers and hashes are serialized in little-endian, and this same notion carries over to Elements.
+
+*Note:* Values are listed in multiples of 100,000,000 (this corresponds directly to how Bitcoin encodes value, where 1 BTC is divisible into 100,000,000 Satoshis). For example, a `ConfidentialAmount` representing 7 inflation keys would be encoded as 700,000,000.
+
+#### Transaction
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Version | Yes | 4 bytes | `int32_t` | Little-endian | Transaction version number. Currently version 2 (see [BIP 68](https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki#specification)). | |
+| Flags | Yes | 1 byte | `unsigned char` | | Whether the transaction contains a witness. Either 0 (no witness) or 1. |
+| Num Inputs | Yes | Varies | `VarInt` | | Number of inputs to the transaction. |
+| Inputs | Yes | Varies | `Vector<TxIn>` | | |
+| Num Inputs | Yes | Varies | `VarInt` | | Number of outputs from the transaction. |
+| Outputs | Yes | Varies | `Vector<TxOut>` | | |
+| Locktime | Yes | 4 bytes | `uint32_t` | Little-endian | See [BIP 113](https://github.com/bitcoin/bips/blob/master/bip-0113.mediawiki). |
+| Witness | Only if flags is 1 | Varies | `Witness` | | See [BIP 141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki). Note that Elements witnesses contain more data than Bitcoin witnesses. This extra data is described further below. |
+
+Serialized format: `[Version][Flags][Num Inputs][Inputs][Num Outputs][Outputs][Locktime][Witness]`
+
+This is the overarching structure of a serialized transaction. The rest of this document contains further details on specific parts, as well as examples.
+
+#### TxIn
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Output Hash (TXID) | Yes | 32 bytes | `uint256` | Little-endian | Bytes of the output hash are encoded in little-endian. **NB**: this means that they appear backwards when serialized. |
```

Comment:
Can probably omit the NB. We can assume familiarity with little-endian and big-endian serializations.

Reply:
FWIW, I think the situation here is more complex than this documentation presents.  The serialization order of the hashes *is* the correct order of the bytes produced by SHA256.  It is the elements client that takes this correct serialization order, reinterprets it as a 256 bit integer in little endian format and then displays that integer in hex, which has the effect of displaying AssetIDs backwards.

I just don't know a concise way of explaining all this.

Reply:
I see. Thank you for the clarification.

I was actually under the (incorrect) impression that the elements client displayed the correct order of bytes produced by SHA256, and the hashes were just serialized in reverse. 

Your explanation makes much more sense.

I will make this distinction more clear.

Reply:
I think we should drop any reference to endianness and just say that the bytes are reversed. You can add a footnote explaining that historically this has something to do with endianness and type conversions but I don't think it would make things any clearer.

Reply:
Done. I've stated that the bytes are reversed, and I've added a footnote explaining the historical reasons for this.